app/services/hyrax/custom_queries/navigators/find_files.rb

```diff
@@ -39,40 +39,48 @@ def find_files(file_set:)
         # @param file_set [Hyrax::FileSet]
         # @return [Hyrax::FileMetadata]
         def find_original_file(file_set:)
-          if file_set.respond_to?(:original_file_id)
-            raise ::Valkyrie::Persistence::ObjectNotFoundError, "File set's original file is blank" if file_set.original_file_id.blank?
-            query_service.custom_queries.find_file_metadata_by(id: file_set.original_file_id)
-          else
-            raise ::Valkyrie::Persistence::ObjectNotFoundError,
-                  "#{file_set.internal_resource} is not a `Hydra::FileSet` implementer"
-          end
+          find_exactly_one_file_by_use(
+            file_set: file_set,
+            use: Hyrax::FileMetadata::Use::ORIGINAL_FILE
+          )
         end
 
         # Find extracted text id of a given file set resource, and map to file metadata resource
         # @param file_set [Hyrax::FileSet]
         # @return [Hyrax::FileMetadata]
         def find_extracted_text(file_set:)
-          if file_set.respond_to?(:extracted_text_id)
-            raise ::Valkyrie::Persistence::ObjectNotFoundError, "File set's extracted text is blank" if file_set.extracted_text_id.blank?
-            query_service.custom_queries.find_file_metadata_by(id: file_set.extracted_text_id)
-          else
-            raise ::Valkyrie::Persistence::ObjectNotFoundError,
-                  "#{file_set.internal_resource} is not a `Hydra::FileSet` implementer"
-          end
+          find_exactly_one_file_by_use(
+            file_set: file_set,
+            use: Hyrax::FileMetadata::Use::EXTRACTED_TEXT
+          )
         end
 
         # Find thumbnail id of a given file set resource, and map to file metadata resource
         # @param file_set [Hyrax::FileSet]
         # @return [Hyrax::FileMetadata]
         def find_thumbnail(file_set:)
-          if file_set.respond_to?(:thumbnail_id)
-            raise ::Valkyrie::Persistence::ObjectNotFoundError, "File set's thumbnail is blank" if file_set.thumbnail_id.blank?
-            query_service.custom_queries.find_file_metadata_by(id: file_set.thumbnail_id)
-          else
-            raise ::Valkyrie::Persistence::ObjectNotFoundError,
-                  "#{file_set.internal_resource} is not a `Hydra::FileSet` implementer"
-          end
+          find_exactly_one_file_by_use(
+            file_set: file_set,
+            use: Hyrax::FileMetadata::Use::THUMBNAIL
+          )
         end
+
+        private
+
+          ##
+          # @api private
+          #
+          # @return [Hyrax::FileMetadata]
+          # @raise [Valkyrie::Persistence::ObjectNotFoundError]
+          def find_exactly_one_file_by_use(file_set:, use:)
+            files =
+              query_service.custom_queries.find_many_file_metadata_by_use(resource: file_set, use: use)
+
+            raise Valkyrie::Persistence::ObjectNotFoundError, "FileSet #{file_set.id}'s #{use.fragment} is missing." if
+              files.blank?
```

Comment:
`#blank?` seems a bit of an odd method to check; I was reading it and expecting to see `#empty?`. Cognitively, I don't think of a "blank" Array. I could also see `unless files.present?` which would read similarly.

Reply:
i was thinking `if files.blank?` (or the equivalent `unless files.present?`) is necessary because some adapters may return `nil` instead of an empty array; but the query called is ours and we can just be strict about its return type.

i'll change to `#empty?`